.github/workflows/main.yml

```diff
@@ -31,43 +37,66 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
-      - name: setup (linux)
-        if: startsWith(matrix.os, 'ubuntu')
-        run: |
-          sudo apt install cmake-data
-
-      # - name: setup (mac)
-      #   if: startsWith(matrix.os, 'macos')
-      #   run: |
-      #     brew update
-      #     brew upgrade
-      #     brew install cmake openssl
 
       - name: setup (windows)
         if: startsWith(matrix.os, 'windows')
         run: |
           choco install openssl cmake
+      
+      - name: setup (linux)
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt install cmake-data
 
       - name: Create Build directory
-        run: cmake -E make_directory ${{runner.workspace}}/build
+        shell: bash
+        run: |
+          pwd
+          echo "$GITHUB_WORKSPACE/build"
+          cmake -E make_directory $GITHUB_WORKSPACE/build
+          cd $GITHUB_WORKSPACE/build
 
       - name: Configure CMake
         shell: bash
-        working-directory: ${{runner.workspace}}/build
         run: |
+          cd $GITHUB_WORKSPACE/build
           cmake $GITHUB_WORKSPACE \
           -DCOSE_C_USE_CONTEXT=${{matrix.USE_CONTEXT}} \
           -DCOSE_C_BUILD_DOCS=OFF \
           -DCOSE_C_COVERALLS=OFF \
           -DCOSE_C_BUILD_TESTS=ON \
           -DBUILD_SHARED_LIBS=OFF \
+          -DCMAKE_BUILD_TYPE=Debug \
           ${{matrix.CMAKE_OPTIONS}} \
           -DCOSE_C_USE_MBEDTLS=${{matrix.USE_MBEDTLS}}
 
       - name: Build
-        working-directory: ${{runner.workspace}}/build
         shell: bash
-        run: cmake --build .
+        run: |
+          cd $GITHUB_WORKSPACE/build
+          cmake --build .
+
+      - name: get test data
+        shell: bash
+        run: |
+          echo $GITHUB_WORKSPACE
+          pwd
+          cd $GITHUB_WORKSPACE
+          pwd
+          git clone --depth 1 git://github.com/cose-wg/Examples Examples
+          ls -al
+
+      - name: Test
+        shell: bash
+        run: |
+          pwd
+          ls -al
+          cd $GITHUB_WORKSPACE
+          pwd
+          ls -al
+          cd $GITHUB_WORKSPACE/build
+          ls -al
+          ctest -C Debug --verbose
 
   clang-tidy:

```

Comment:
This does not end in the test step - should it?  I seem to remember that this is where gcc is run and above is where clang is run.

Reply:
do not get it, sorry...
were we test with the default compiler e.g. gcc on ubuntu and visual studio on windows...

below we also run clang-tidy static analysis, there we use clang as the compiler

github also offers macos, this could be added in a next step...

Reply:
Currently we run the tests for both the clang compiler and the gcc compiler as well s the Msft compiler for a small set.  With this change we run tests only for gcc and Msft.  I don't know that this would be an issue as running should be the same unless there is some strange things in the runtime.  That is not necessarily true with the Msft compiler.